### PR TITLE
Issue #250 | Using patched version of avro

### PR DESCRIPTION
**Pull request by druminski**

### CODE REVIEW

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/Topic.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public RetentionTime getRetentionTime() {
     @Override
     public int hashCode() {
         return Objects.hash(name, description, retentionTime, messageSchema, validationEnabled, validationDryRunEnabled,
-                migratedFromJsonType, trackingEnabled, ack, contentType);
+                migratedFromJsonType, trackingEnabled, ack, contentType, jsonToAvroDryRunEnabled);
```
Comment (druminski) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/Topic.java]: small fix not related to this issue

File: hermes-common/build.gradle [hunk 2]
```diff
@@ -22,7 +22,16 @@ dependencies {
...
+    /**
+     * Patched version contains two fixes:
+     *
+     * https://issues.apache.org/jira/browse/AVRO-1582
+     * https://issues.apache.org/jira/browse/AVRO-1717
+     *
+     * They improve conversion of json to avro.
+     * Patched version will be used until mentioned fixes are released in official version of avro.
+     */
+    compile files('./lib/avro-1.8.0-json2avroPatch-SNAPSHOT.jar')
```
Comment (pbetkier) [inline hermes-common/build.gradle]: Nice explanation.

**Comment by pbetkier** [comment]
:+1:

**Comment by dankraw** [comment]
:+1:

File: hermes-common/src/test/java/pl/allegro/tech/hermes/common/message/converter/JsonToAvroConverterTest.java [hunk 3]
```diff
@@ -31,6 +31,9 @@ public void shouldConvertToAvro() throws IOException {
         BinaryDecoder decoder = DecoderFactory.get().binaryDecoder(avro, null);
         assertThat(decoder.readString()).isEqualTo("Bob");
         assertThat(decoder.readInt()).isEqualTo(50);
+        //next filed is type of union: [null, string]
```
Comment (wojtkiewicz) [inline hermes-common/src/test/java/pl/allegro/tech/hermes/common/message/converter/JsonToAvroConverterTest.java]: typo: field

**Comment by wojtkiewicz** [comment]
:+1:

**Pull request merged**